src/app-layout/visual-refresh-toolbar/multi-layout.ts

```diff
@@ -69,15 +69,34 @@ function mergeProps(ownProps: SharedProps, additionalProps: ReadonlyArray<Partia
   return Object.keys(toolbar).filter(key => key !== 'ariaLabels').length > 0 ? toolbar : null;
 }
 
-export function useMultiAppLayout(props: SharedProps) {
+export function useMultiAppLayout(props: SharedProps, ref: RefObject<HTMLDivElement>) {
   const [registration, setRegistration] = useState<RegistrationState<SharedProps> | null>(null);
   const { forceDeduplicationType } = props;
 
   useLayoutEffect(() => {
-    return awsuiPluginsInternal.appLayoutWidget.register(forceDeduplicationType, props =>
-      setRegistration(props as RegistrationState<SharedProps>)
-    );
-  }, [forceDeduplicationType]);
+    if (!ref.current) {
+      return;
+    }
+    let unregister: () => void | undefined;
+
+    const observer = new IntersectionObserver(([entry]) => {
```

Comment:
Our `useIntersectionObserver` hook is not suitable here, because it initially (on the first render) sets visibility state to `false` and then, after first rerender, sets it to true, while here we can initially set the correct value

Reply:
Update: updated the shared hook to make use of it here